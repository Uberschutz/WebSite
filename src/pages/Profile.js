import React, { Component } from 'react';
import '../styles/bootstrap.css';

import { Icon } from 'antd';

import {displayContent} from '../utils/translationDisplay';
import Unauthorized from "./Unauthorized";

class Profile extends Component {
    render() {
    	if (this.props.logged) {
		    let i = 0;
		    return (
			    <div className="description-txt">
				    <h4 className="name-font">
					    {displayContent(this.props.lang, i++, 'profile')}
				    </h4> <br/>
				    <h6>
					    {displayContent(this.props.lang, i++, 'profile')}<br/><br/>
					    {displayContent(this.props.lang, i++, 'profile')}<br/><br/>
					    {displayContent(this.props.lang, i++, 'profile')}<br/>
				    </h6> <br/>
				    <table className="table">
					    <thead className="table-primary">
					    <tr>
						    <th scope="col">Date</th>
						    <th scope="col">{displayContent(this.props.lang, i++, 'profile')}</th>
						    <th scope="col">{displayContent(this.props.lang, i++, 'profile')}</th>
					    </tr>
					    </thead>
					    <tbody>
					    <tr>
						    <th scope="row">22/05/2019</th>
						    <td>Achat d'Überschutz Premium</td>
						    <td>
							    <Icon type="check-circle" theme="twoTone"
							          twoToneColor="#52c41a"/>
						    </td>
					    </tr>
					    <tr>
						    <th scope="row">25/05/2019</th>
						    <td>Prélèvement n°1 Überschutz Premium</td>
						    <td>En cours ...</td>
					    </tr>
					    <tr>
						    <th scope="row">25/06/2019</th>
						    <td>Prélèvement n°2 Überschutz Premium</td>
						    <td>Prochainement</td>
					    </tr>
					    </tbody>
				    </table>
				    <br/>
			    </div>
		    )
	    } else {
    		return (
    			<Unauthorized/>
		    )
	    }
    }
}

export default Profile;