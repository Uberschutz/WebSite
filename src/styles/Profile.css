.txt-align {
    align-items: center!important;
    align-content: center;
    justify-content: center;
    justify-items: center;
    text-align: center;
}

.description-txt {
    text-align: left!important;
}

.name-font {
    font-size: 120%;
}

.button-footerP {
    margin-top: 1.5%;
    margin-bottom: 1.5%;
}

.right-btn {
    text-align: right!important;
    margin-right: 1%;
}

.fixed-size {
    overflow: hidden;
}

.options-margin {
    margin-top: 3%;
    margin-bottom: 3%;
}
