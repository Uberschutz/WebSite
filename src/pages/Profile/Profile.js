import React, { Component } from 'react';
import '../../styles/bootstrap.css';
import '../../styles/Profile.css';

import { Icon } from 'antd';
import {displayContent} from '../../utils/translationDisplay';
import { Alert } from 'reactstrap';

import Unauthorized from "../Unauthorized";
import newsletter from "../../assets/icons8-mail-100.png";
import loading from '../../assets/Spinner-1s-70px.gif';

import axios from 'axios';

import fileDownload from "js-file-download";

class Profile extends Component {
	constructor(props) {
		super(props);

		this.state = {
			lang: 'fr',
			logged: false,
			email: '',
			firstname: '',
			lastname: '',
			newsletter: false,
			subscribed: false,
			subscription: '',
			token: null,
			load: false,
			newsletterAlert: '',
			statusErr: false
		};
		// this._handleKeyPressed = this._handleKeyPressed.bind(this);
		this.onChangeFirstname = this.onChangeFirstname.bind(this);
		this.onChangeLastname = this.onChangeLastname.bind(this);
		this.changeFirstName = this.changeFirstName.bind(this);
		this.changeLastName = this.changeLastName.bind(this);
		this.getAccountData = this.getAccountData.bind(this);
		this.deleteAccount = this.deleteAccount.bind(this);
		this.unsubscribe_newsletter = this.unsubscribe_newsletter.bind(this);
		this.getUser = this.getUser.bind(this);
	}

	componentDidMount() {
		if (this.props.base) {
			const { base: { language, logged } } = this.props;
			this.setState({
				logged, language
			}, this.getUser);
		// 	axios.get('/get_subscription', {
		// 		headers: {
		// 			'x-access-token': token
		// 		}
		// 	}).then(response => {
		// 		this.setState({
		// 			lang: language,
		// 			logged : logged,
		// 			token,
		// 			newsletter,
		// 			email,
		// 			lastname,
		// 			firstname,
		// 			subscription: response.data
		// 		});
		// 	}).catch(err => {
		// 		console.log(err);
		// 		this.setState({
		// 			lang: language,
		// 			logged : logged,
		// 			token,
		// 			newsletter,
		// 			email,
		// 			lastname,
		// 			firstname
		// 		});
		// 	})
		}
	}

	componentDidUpdate(prevProps, prevState, snapshot) {
		// console.log(prevProps, this.props);
		this.props.base !== prevProps.base && this.setState({lang: this.props.base.language, logged: this.props.base.logged}, () => console.log('re'));
	}

	getUser() {
		if (this.props.base && this.props.base.token) {
			axios.get('/get_auth_user', {
				headers: {
					token: this.props.base.token
				}
			}).then(response => {
				if (response && response.data) {
					this.setState({
						firstname: response.data.firstname,
						lastname: response.data.lastname,
						email: response.data.email,
						newsletter: response.data.newsletter,
						subscription: response.data.subscription
					});
				}
			}).catch(err => {
				console.log(err);
			})
		}
	}

	unsubscribe_newsletter() {
		this.setState({
			load: true
		}, () => {
				axios.post('/unsubscribe_newsletter', {
			}, {
				headers: {
					'x-access-token': this.state.token
				}
			}).then(response => {
				console.log(response.data);
				this.setState({
					newsletter: false,
					load: false,
					statusErr: false,
					newsletterAlert: 'Successfully unsubscribed to the newsletter'
				});
				this.props.setNewsletter(false);
			}).catch(err => {
				this.setState({
					load: false,
					statusErr: true,
					newsletterAlert: 'An error occurred while unsubscribing your account to the newsletter'
				});
				console.log(err);
			}).finally(() => {
					setTimeout(() => {
						this.setState({
							newsletterAlert: ''
						})
					}, 1000 * 5)
			});
		});
	}

	changeLastName() {
		axios.post('/rename', {
			key: 'lastname',
			value: this.state.lastname
		},{
			headers: {
				'x-access-token': this.state.token
			}
		}).then(response => {
			this.props.setUser(this.state.email, this.state.lastname, this.state.firstname, this.state.token);
		}).catch(err => {
			console.log(err, "Error");
		})
	}

	changeFirstName() {
		axios.post('/rename', {
			key: 'firstname',
			value: this.state.firstname
		},{
			headers: {
				'x-access-token': this.state.token
			}
		}).then(response => {
			this.props.setUser(this.state.email, this.state.lastname, this.state.firstname, this.state.token);
		}).catch(err => {
			console.log(err, "Error");
		})
	}

	onChangeFirstname(value) {
		this.setState({
			firstname: value.target.value
		})
	}

	onChangeLastname(value) {
		this.setState({
			lastname: value.target.value
		})
	}

	deleteAccount() {
		axios.post('/delete_account', '',{
			headers: {
				'x-access-token': this.state.token
			}
		}).then(response => {
			this.props.setLogged(false);
			this.props.setUser(null, null, null, null);
			this.props.history.push("/");
		}).catch(err => {
			console.log(err, "Error");
		})
	}

	getAccountData() {
		axios.get('/gdpr', {
			headers: {
				'x-access-token': this.state.token
			}
		}).then(response => {
			try {
				fileDownload(JSON.stringify(response.data), 'data.json');
			} catch (e) {
				console.log(e);
			}
		}).catch(err => {
			console.log(err);
		})
	}

	_handleKeyPressed(e, action) {
		if (e.key === "Enter") {
			if (action === "firstname") {
				this.changeFirstName();
			} else {
				this.changeLastName();
			}
		}
	}

    render() {
		if (this.state.logged) {
		    let i = 0;
		    return (
			    <div className="fixed-size description-txt">
					<div className="row button-footerP">
						<h4 className="name-font col-sm-auto">
							{displayContent(this.state.lang, i++, 'profile')}
						</h4>
						<div className="input-group input-group-sm col-sm-2">
							<input type="text" className="form-control" aria-label="Small" aria-describedby="inputGroup-sizing-sm" onKeyPress={(e) => { this._handleKeyPressed(e, "lastname")}} value={this.state.lastname} onChange={this.onChangeLastname}/>
							<button className="btn btn-primary btn-sm" onClick={this.changeLastName}>OK</button>
						</div> <br/>
					</div>
					<div className="row button-footerP">
						<h5 className="name-font col-sm-auto">
							{displayContent(this.state.lang, i++, 'profile')}
						</h5>
						<div className="input-group input-group-sm col-sm-2">
							<input type="text" className="form-control" aria-label="Small" aria-describedby="inputGroup-sizing-sm" onKeyPress={(e) => { this._handleKeyPressed(e, "firstname")}} value={this.state.firstname} onChange={this.onChangeFirstname}/>
							<button className="btn btn-primary btn-sm" onClick={this.changeFirstName}>OK</button>
						</div><br/>
					</div>
					<h6>
						{displayContent(this.state.lang, i++, 'profile')}<br/><br/>
						{displayContent(this.state.lang, i++, 'profile')}
						<span>{this.state.subscription}</span>
						{/*Ecrire achat Üz ici*/}
					</h6>
					{
					++i && this.state.newsletter ?
						<div>
							<h6 className="right-btn">
								Newsletter : <button className="btn btn-dark btn-sm" onClick={this.unsubscribe_newsletter}>
								<img src={newsletter} alt="newsletter"/>{displayContent(this.state.lang, i - 1, 'profile')} {
								this.state.load ? <img src={loading} alt="loading"/> : null
							}
							</button>
							</h6>
							{
								this.state.newsletterAlert !== '' ? <Alert
									color={this.state.statusErr ? "danger" : "success"}>{this.state.newsletterAlert}</Alert> : null
							}
						</div> : null
					}
					<h6>
						{displayContent(this.state.lang, i++, 'profile')}
					</h6><br/>
				    <table className="table">
					    <thead className="table-primary">
					    <tr>
						    <th scope="col">Date</th>
						    <th scope="col">{displayContent(this.state.lang, i++, 'profile')}</th>
						    <th scope="col">{displayContent(this.state.lang, i++, 'profile')}</th>
					    </tr>
					    </thead>
					    <tbody>
					    <tr>
						    <th scope="row">22/05/2019</th>
						    <td>Achat d'Überschutz Premium</td>
						    <td>
							    <Icon type="check-circle" theme="twoTone"
							          twoToneColor="#52c41a"/>
						    </td>
					    </tr>
					    <tr>
						    <th scope="row">25/05/2019</th>
						    <td>Prélèvement n°1 Überschutz Premium</td>
						    <td>En cours ...</td>
					    </tr>
					    <tr>
						    <th scope="row">25/06/2019</th>
						    <td>Prélèvement n°2 Überschutz Premium</td>
						    <td>Prochainement</td>
					    </tr>
					    </tbody>
				    </table>
				    <br/>
				    <div className="row txt-align">
						<button className="col-2 btn btn-primary button-footerP" onClick={this.getAccountData}>{displayContent(this.state.lang, i++, 'profile')}</button>
						<button className="col-2 btn btn-danger button-footerP" onClick={this.deleteAccount}>{displayContent(this.state.lang, i++, 'profile')}</button>
					</div>
			    </div>
		    )
	    } else {
    		return (
    			<Unauthorized/>
		    )
	    }
    }
}

export default Profile;