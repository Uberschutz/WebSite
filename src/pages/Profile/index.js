import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import * as actions from '../../actions'
import Profile from './Profile'

const mapStateToProps = state => ({
	base: state.base,
})

const mapDispatchToProps = dispatch => ({

})

const ProfileContainer = connect(mapStateToProps, mapDispatchToProps)(Profile)

export default ProfileContainer