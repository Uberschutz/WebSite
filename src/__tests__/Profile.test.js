import React from 'react';
import ReactDOM from 'react-dom';
import Profile from '../pages/Profile/Profile';

it('renders Profile without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Profile lang='fr' />, div);
    expect(div).toMatchSnapshot();
    ReactDOM.unmountComponentAtNode(div);
});
