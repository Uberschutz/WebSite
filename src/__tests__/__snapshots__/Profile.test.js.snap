// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Profile page rendering renders Profile being logged in english 1`] = `
<div>
  <div
    class="fixed-size"
  >
    <div
      class="description-txt"
    >
      <div
        class="row button-footerP"
      >
        <h4
          class="name-font col-sm-auto"
        >
          Nom de famille : 
        </h4>
        <div
          class="input-group input-group-sm col-sm-2"
        >
          <input
            aria-describedby="inputGroup-sizing-sm"
            aria-label="Small"
            class="form-control"
            data-action="lastname"
            type="text"
            value=""
          />
          <button
            class="btn btn-primary btn-sm"
          >
            OK
          </button>
        </div>
         
        <br />
      </div>
      <div
        class="row button-footerP"
      >
        <h5
          class="name-font col-sm-auto"
        >
          Prénom : 
        </h5>
        <div
          class="input-group input-group-sm col-sm-2"
        >
          <input
            aria-describedby="inputGroup-sizing-sm"
            aria-label="Small"
            class="form-control"
            data-action="firstname"
            type="text"
            value=""
          />
          <button
            class="btn btn-primary btn-sm"
          >
            OK
          </button>
        </div>
        <br />
      </div>
      <h6>
        Enfants protégés par Überschutz : X 
        <br />
        <br />
        Abonnement utilisé : 
        <span />
      </h6>
      <h6>
        Echéancier de paiement : 
      </h6>
      <br />
      <table
        class="table"
      >
        <thead
          class="table-primary"
        >
          <tr>
            <th
              scope="col"
            >
              Date
            </th>
            <th
              scope="col"
            >
              Nom du prélèvement
            </th>
            <th
              scope="col"
            >
              Payé ?
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th
              scope="row"
            >
              22/05/2019
            </th>
            <td>
              Achat d'Überschutz Premium
            </td>
            <td>
              <i
                aria-label="icon: check-circle"
                class="anticon anticon-check-circle"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="check-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                    fill="#52c41a"
                  />
                  <path
                    d="M512 140c-205.4 0-372 166.6-372 372s166.6 372 372 372 372-166.6 372-372-166.6-372-372-372zm193.4 225.7l-210.6 292a31.8 31.8 0 0 1-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.3 0 19.9 5 25.9 13.3l71.2 98.8 157.2-218c6-8.4 15.7-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.4 12.7z"
                    fill="#f6ffed"
                  />
                  <path
                    d="M699 353h-46.9c-10.2 0-19.9 4.9-25.9 13.3L469 584.3l-71.2-98.8c-6-8.3-15.6-13.3-25.9-13.3H325c-6.5 0-10.3 7.4-6.5 12.7l124.6 172.8a31.8 31.8 0 0 0 51.7 0l210.6-292c3.9-5.3.1-12.7-6.4-12.7z"
                    fill="#52c41a"
                  />
                </svg>
              </i>
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            >
              25/05/2019
            </th>
            <td>
              Prélèvement n°1 Überschutz Premium
            </td>
            <td>
              En cours ...
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            >
              25/06/2019
            </th>
            <td>
              Prélèvement n°2 Überschutz Premium
            </td>
            <td>
              Prochainement
            </td>
          </tr>
        </tbody>
      </table>
      <br />
    </div>
    <button
      class="btn btn-outline-dark options-margin"
      type="button"
    >
      Modifier mon adresse email
    </button>
    <button
      class="btn btn-outline-dark options-margin"
      type="button"
    >
      Modifier mon mot de passe
    </button>
    <div
      class="row txt-align"
    >
      <button
        class="col-2 btn btn-primary options-margin"
      >
        Récupérer mes informations
      </button>
      <button
        class="col-2 btn btn-danger options-margin"
      >
        Supprimer mon compte
      </button>
    </div>
  </div>
</div>
`;

exports[`Testing the Profile page rendering renders Profile being logged in french 1`] = `
<div>
  <div
    class="fixed-size"
  >
    <div
      class="description-txt"
    >
      <div
        class="row button-footerP"
      >
        <h4
          class="name-font col-sm-auto"
        >
          Nom de famille : 
        </h4>
        <div
          class="input-group input-group-sm col-sm-2"
        >
          <input
            aria-describedby="inputGroup-sizing-sm"
            aria-label="Small"
            class="form-control"
            data-action="lastname"
            type="text"
            value=""
          />
          <button
            class="btn btn-primary btn-sm"
          >
            OK
          </button>
        </div>
         
        <br />
      </div>
      <div
        class="row button-footerP"
      >
        <h5
          class="name-font col-sm-auto"
        >
          Prénom : 
        </h5>
        <div
          class="input-group input-group-sm col-sm-2"
        >
          <input
            aria-describedby="inputGroup-sizing-sm"
            aria-label="Small"
            class="form-control"
            data-action="firstname"
            type="text"
            value=""
          />
          <button
            class="btn btn-primary btn-sm"
          >
            OK
          </button>
        </div>
        <br />
      </div>
      <h6>
        Enfants protégés par Überschutz : X 
        <br />
        <br />
        Abonnement utilisé : 
        <span />
      </h6>
      <h6>
        Echéancier de paiement : 
      </h6>
      <br />
      <table
        class="table"
      >
        <thead
          class="table-primary"
        >
          <tr>
            <th
              scope="col"
            >
              Date
            </th>
            <th
              scope="col"
            >
              Nom du prélèvement
            </th>
            <th
              scope="col"
            >
              Payé ?
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th
              scope="row"
            >
              22/05/2019
            </th>
            <td>
              Achat d'Überschutz Premium
            </td>
            <td>
              <i
                aria-label="icon: check-circle"
                class="anticon anticon-check-circle"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="check-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                    fill="#52c41a"
                  />
                  <path
                    d="M512 140c-205.4 0-372 166.6-372 372s166.6 372 372 372 372-166.6 372-372-166.6-372-372-372zm193.4 225.7l-210.6 292a31.8 31.8 0 0 1-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.3 0 19.9 5 25.9 13.3l71.2 98.8 157.2-218c6-8.4 15.7-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.4 12.7z"
                    fill="#f6ffed"
                  />
                  <path
                    d="M699 353h-46.9c-10.2 0-19.9 4.9-25.9 13.3L469 584.3l-71.2-98.8c-6-8.3-15.6-13.3-25.9-13.3H325c-6.5 0-10.3 7.4-6.5 12.7l124.6 172.8a31.8 31.8 0 0 0 51.7 0l210.6-292c3.9-5.3.1-12.7-6.4-12.7z"
                    fill="#52c41a"
                  />
                </svg>
              </i>
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            >
              25/05/2019
            </th>
            <td>
              Prélèvement n°1 Überschutz Premium
            </td>
            <td>
              En cours ...
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            >
              25/06/2019
            </th>
            <td>
              Prélèvement n°2 Überschutz Premium
            </td>
            <td>
              Prochainement
            </td>
          </tr>
        </tbody>
      </table>
      <br />
    </div>
    <button
      class="btn btn-outline-dark options-margin"
      type="button"
    >
      Modifier mon adresse email
    </button>
    <button
      class="btn btn-outline-dark options-margin"
      type="button"
    >
      Modifier mon mot de passe
    </button>
    <div
      class="row txt-align"
    >
      <button
        class="col-2 btn btn-primary options-margin"
      >
        Récupérer mes informations
      </button>
      <button
        class="col-2 btn btn-danger options-margin"
      >
        Supprimer mon compte
      </button>
    </div>
  </div>
</div>
`;

exports[`Testing the Profile page rendering renders Profile not being logged in english 1`] = `
<div>
  <div>
    <h1
      class="text-xl-center forbidden"
    >
      403 Accès interdit
    </h1>
    <br />
    <img
      alt="forbidden"
      src="fogg-no-comments.png"
    />
  </div>
</div>
`;

exports[`Testing the Profile page rendering renders Profile not being logged in french 1`] = `
<div>
  <div>
    <h1
      class="text-xl-center forbidden"
    >
      403 Accès interdit
    </h1>
    <br />
    <img
      alt="forbidden"
      src="fogg-no-comments.png"
    />
  </div>
</div>
`;

exports[`Testing the Profile page rendering renders Profile without crashing 1`] = `
<div>
  <div>
    <h1
      class="text-xl-center forbidden"
    >
      403 Accès interdit
    </h1>
    <br />
    <img
      alt="forbidden"
      src="fogg-no-comments.png"
    />
  </div>
</div>
`;
